js/util.ts

```diff
@@ -125,3 +125,26 @@ export function deferred(): Deferred {
     reject: reject!
   };
 }
+
+/** Create a IterableIterator. */
+// @internal
+export class CreateIterableIterator<T> implements IterableIterator<T> {
+  private readonly _data: T[];
+  private index = 0;
+  readonly [Symbol.toStringTag] = "Iterator";
+
+  constructor(data: T[]) {
```

Comment:
Or generalize it by having it accept `Iterable<T>` and then call the iterator:
```js
constructor(data: Iterable<T>) {
  this._data = data[Symbol.iterator]();
}
```
...which may or may not be a no-op but then this could accept any iterable thing.

Reply:
Awesome! Updated it.